k8s/k8s.go

```diff
@@ -0,0 +1,291 @@
+					}
+				}
+				for _, r := range v.Ranges {
+					for i := 1; i < len(r.Events); i++ {
+						if len(r.Events[i-1].Introduced) == 0 {
+							result = errors.Join(result, fmt.Errorf("\nAffectedVersion Introduced is missing from cve #%s", cve.ID))
+						}
+						if len(r.Events[i].Fixed) == 0 && len(r.Events[i].LastAffected) == 0 {
+							result = errors.Join(result, fmt.Errorf("\nAffectedVersion Fixed and LastAffected are missing from cve #%s", cve.ID))
+						}
+					}
+				}
+			}
+		}
+		if len(cve.References) == 0 {
+			result = errors.Join(result, fmt.Errorf("\nUrls is mssing on cve #%s", cve.ID))
+		}
+	}
+	return result
+}
+
+func cveMissingImportantData(vulnerability *Cve) bool {
+	return len(vulnerability.versions) == 0 ||
+		len(vulnerability.Package) == 0 ||
+		len(vulnerability.CvssV3.Vector) == 0
+}
+
+// getExitingCvesToModifiedMap get the existing cves from vuln-list-k8s repo and map it to cve id and last updated
+func getExitingCvesToModifiedMap() (map[string]string, error) {
+	response, err := http.Get(vulnListRepoTarBall)
+	if err != nil {
+		return nil, err
+	}
+	defer response.Body.Close()
+	return tarToMap(response.Body)
+}
+
+// tarToMap read ewxisting cves from tar file and map it to cve id and last updated
+func tarToMap(reader io.ReadCloser) (map[string]string, error) {
+	cveTimeMap := make(map[string]string)
+	gz, err := gzip.NewReader(reader)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		_ = gz.Close()
+	}()
+	tarReader := tar.NewReader(gz)
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return nil, err
+		}
+		info := header.FileInfo()
+		switch header.Typeflag {
+
+		// if its a dir and it doesn't exist create it
+		case tar.TypeDir:
+			continue
+		case tar.TypeReg:
+			if strings.Contains(info.Name(), "CVE-") && strings.HasSuffix(info.Name(), ".json") {
+				b, err := io.ReadAll(tarReader)
+				if err != nil {
+					return nil, err
+				}
+				var cve osv.OSV
+				err = json.Unmarshal([]byte(strings.ReplaceAll(string(b), "\n", "")), &cve)
+				if err != nil {
+					return nil, err
+				}
+				cveTimeMap[cve.ID] = cve.Modified
+			}
+		}
+	}
+	return cveTimeMap, nil
+}
+
+// olderCve check if the current cve is older than the existing one
+func olderCve(cveID string, currentCVEUpdated string, existCveLastUpdated map[string]string) bool {
+	if len(existCveLastUpdated) == 0 {
+		return false
+	}
+	if lastUpdated, ok := existCveLastUpdated[cveID]; ok {
+		existastUpdated, err := time.Parse(time.RFC3339, lastUpdated)
+		if err != nil {
+			return false
+		}
+		currentLastUpdated, err := time.Parse(time.RFC3339, currentCVEUpdated)
+		if err != nil {
+			return false
+		}
+		// check if the current collcted cve is older or same as the existing one
+		if currentLastUpdated.Before(existastUpdated) || currentLastUpdated == existastUpdated {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
nit: I prefer aligning the happy path to the left edge.
https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88


```suggestion
	if lastUpdated, ok := existCveLastUpdated[cveID]; !ok {
		return false
	}
	
	existastUpdated, err := time.Parse(time.RFC3339, lastUpdated)
	if err != nil {
		return false
	}
	currentLastUpdated, err := time.Parse(time.RFC3339, currentCVEUpdated)
	if err != nil {
		return false
	}
	// check if the current collcted cve is older or same as the existing one
	if currentLastUpdated.Before(existastUpdated) || currentLastUpdated == existastUpdated {
		return true
	}
	return false
}
```